channels.nix

```diff
@@ -1,87 +1,122 @@
 rec {
   channels = {
-      # "Channel name" = {
-      #   job = "project/jobset/jobname"; like https://hydra.nixos.org/job/<value>/latest-finished
-      #   # when adding a new version, mark the oldest as "not current". New releases should always be current.
-      #   # Channels where `current = false` are marked `end-of-life` on https://status.nixos.org/ and alerting is disabled.
-      #   current = true;
-      # };
-      "nixos-unstable" = {
-        job = "nixos/trunk-combined/tested";
-        current = true;
-      };
-      "nixos-unstable-small" = {
-        job = "nixos/unstable-small/tested";
-        current = true;
-      };
-      "nixpkgs-unstable" = {
-        job = "nixpkgs/trunk/unstable";
-        current = true;
-      };
-      
-      "nixos-21.11" = {
-        job = "nixos/release-21.11/tested";
-        current = true;
-      };
-      "nixos-21.11-small" = {
-        job = "nixos/release-21.11-small/tested";
-        current = true;
-      };
-      "nixpkgs-21.11-darwin" = {
-        job = "nixpkgs/nixpkgs-21.11-darwin/darwin-tested";
-        current = true;
-      };
-      "nixos-21.11-aarch64" = {
-        job = "nixos/release-21.11-aarch64/tested";
-        current = true;
-      };
+    # "Channel name" = {
+    #   # This should be the <value> part of
+    #   # https://hydra.nixos.org/job/<value>/latest-finished
+    #   job = "project/jobset/jobname"; 
+    #
+    #   # When adding a new version, determine if it needs to be tagged as a
+    #   # variant -- for example:
+    #   # nixos-xx.xx         => primary
+    #   # nixos-xx.xx-small   => small
+    #   # nixos-xx.xx-darwin  => darwin
+    #   # nixos-xx.xx-aarch64 => aarch64
+    #   variant = "primary";
+    #
+    #   # Channel Status:
+    #   # '*-unstable' channels are always "rolling"
+    #   # Otherwise a release generally progresses through the following phases:
+    #   #
+    #   #  - Directly after branch off                   => "beta"
+    #   #  - Once the channel is released                => "stable"
+    #   #  - Once the next channel is released           => "deprecated"
+    #   #  - N months after the next channel is released => "unmaintained"
+    #   #    (check the release notes for when this should happen)
```

Comment:
While this *should* be one month, this might change down the road. The source of truth for that specific release should be its release notes.